apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -0,0 +1,305 @@
+	// flavorsUsage are the used quotas, by flavor, currently in use by the
+	// workloads assigned to this ClusterQueue.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	FlavorsUsage []FlavorUsage `json:"flavorsUsage"`
+
+	// pendingWorkloads is the number of workloads currently waiting to be
+	// admitted to this clusterQueue.
+	// +optional
+	PendingWorkloads int32 `json:"pendingWorkloads"`
+
+	// admittedWorkloads is the number of workloads currently admitted to this
+	// clusterQueue and haven't finished yet.
+	// +optional
+	AdmittedWorkloads int32 `json:"admittedWorkloads"`
+
+	// conditions hold the latest available observations of the ClusterQueue
+	// current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type FlavorUsage struct {
+	// name of the flavor.
+	Name string `json:"name"`
+
+	// resources lists the quota usage for the resources in this flavor.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=16
+	Resources []ResourceUsage `json:"resources"`
+}
+
+type ResourceUsage struct {
+	// name of the resource
+	Name corev1.ResourceName `json:"name"`
+
+	// total is the total quantity of used quota, including the amount borrowed
+	// from the cohort.
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Borrowed is quantity of quota that is borrowed from the cohort. In other
+	// words, it's the used quota that is over the nominalQuota.
+	Borrowed resource.Quantity `json:"borrowed,omitempty"`
+}
+
+const (
+	// ClusterQueueActive indicates that the ClusterQueue can admit new workloads and its quota
+	// can be borrowed by other ClusterQueues in the same cohort.
+	ClusterQueueActive string = "Active"
+)
+
+type PreemptionPolicy string
+
+const (
+	PreemptionPolicyNever         PreemptionPolicy = "Never"
+	PreemptionPolicyAny           PreemptionPolicy = "Any"
+	PreemptionPolicyLowerPriority PreemptionPolicy = "LowerPriority"
+)
+
+// ClusterQueuePreemption contains policies to preempt Workloads from this
+// ClusterQueue or the ClusterQueue's cohort.
+type ClusterQueuePreemption struct {
+	// reclaimWithinCohort determines whether a pending Workload can preempt
+	// Workloads from other ClusterQueues in the cohort that are using more than
+	// their min quota. Possible values are:
+	//
+	// - `Never` (default): do not preempt workloads in the cohort.
+	// - `LowerPriority`: if the pending workload fits within the min
+	//   quota of its ClusterQueue, only preempt workloads in the cohort that have
+	//   lower priority than the pending Workload.
+	// - `Any`: if the pending workload fits within the min quota of its
+	//   ClusterQueue, preempt any workload in the cohort, irrespective of
+	//   priority.
+	//
+	// +kubebuilder:default=Never
+	// +kubebuilder:validation:Enum=Never;LowerPriority;Any
+	ReclaimWithinCohort PreemptionPolicy `json:"withinCohort,omitempty"`
```

Comment:
```suggestion
	WithinCohort PreemptionPolicy `json:"withinCohort,omitempty"`
```
In order to align the struct name and the json field name.

Reply:
Oops, this was a mistake in #514. It should be `reclaimWithinCohort`